tests/foreman/cli/test_remoteexecution.py

```diff
@@ -100,6 +82,24 @@ def infra_host(request, target_sat, module_capsule_configured):
     yield infra_hosts[request.param]
 
 
+def assert_job_invocation(invocation_command_id, client_hostname, expected_result='success'):
```

Comment:
you can keep expected_result='1' and `assert result['success'] == expected_result` no need for if else block.

Reply:
that would be nice, but the annoyance here is that I need to check `result['stauts']` instead of `result['success']`

Reply:
I think you're right you'll still need an if block here, but there's a bit of maybe mis-naming going on?

`expected_result` is the variable, but is this really the expected result, or is it the expected status?

Is `status` still set once `success` is set? (leading question, can you assert status still if success is expected?)

The kwarg is a string, but only has two expected states.  It can't be called with an expected status of 'failed', or 'in progress', or anything else - so maybe it should just be a boolean `assert_success` kwarg?  Seems ugly to take `field` and `status` as separate kargs for the caller to indicate which field and what state.

Reply:
Nice! Yeah this cleans up the tests very well and makes for clear separation of intent in what's being called.